### DISCUSSION TITLE
bTicino  K4003C can't control the led

### DISCUSSION THREAD

**Discussion by andreapx** [Koenkk/zigbee2mqtt#17597] [Devices]
Hi,
I'm testing the  K4003C/L4003C/N4003C/NT4003C switch. ON and OFF works without problems, but I can't get/set the led.
I'm sending these commands, but I don't receive anything back and the led is always on:
```
{"led_if_on": ""}
{"led_in_dark": ""}
{"led_if_on": "ON"}
{"led_if_on": "OFF"}
{"led_in_dark": "ON"}
{"led_in_dark": "OFF"}
```

Am I doing something wrong?

![immagine](https://user-images.githubusercontent.com/13846728/236757685-047859fe-ad12-4bdc-b1a6-7a53acf371ca.png)

**Comment by andreapx** [comment]
I've checked and the state of the led in the device is off:
`{"brightness":100,"device_mode":"dimmer_on","led_in_dark":"OFF","led_when_off":"OFF","led_when_on":"OFF","linkquality":142,"state":"OFF"}`

Has anybody else had this problem?

**Reply by cr4sh0verride** [reply]
Yep I have exactly the same issue
Doesn't matter what I send the device does not take that setting 

Let me know if you find an answer

**Reply by andreapx** [reply]
Unfortunately I haven't solved it yet.

**Reply by cr4sh0verride** [reply]
The payload it sends mine is 

{"ballast_lamp_alarm_lamp_burn_hours":false,"ballast_maximum_level":254,"ballast_minimum_level":1,"ballast_status_lamp_failure":false,"ballast_status_non_operational":false,"brightness":1,"device_mode":"dimmer_on","led_if_on":"OFF","led_in_dark":"OFF","led_when_off":"OFF","linkquality":255,"state":"ON"}'

I noticed that 
"led_when_off":"OFF"
Is not part of any command you can send to it

I tried to set it manually in the state.json file but whenever it sends a command it just sends the OFF
Not sure if that would have something to do with it

I noticed the behaviour when it's connect to the Home App as well and the light will turn off
You can set it to be on when the light is off.. but the light is solid blue and not green

So I think solid green has something to do with it's firmware and not connecting to something specifically in the Home Gateway it usually connects to (outside of the custom zigbee gateway device)

**Reply by cr4sh0verride** [reply]
I fixed it 
I found it in the notes of the  toZigbee.js convertor file 

enable or disable the LED (blue) when permitJoin=false (LED off)

I had permanent join on in the config file
I turned it off and all the LEDS turned off now

**Comment by pinowet** [comment]
Hi cr4sh0verride,
irst of all merry Christmas :) 

I have the same problem as you, that the 4003 switches remain with the steady green light even if I turn off the pairing on zigbee2MQTT, but I don't really understand how you fixed the problem.
Is there a particular procedure to be done or a file to be modified?

Thank you
